apic_ml2/neutron/plugins/ml2/drivers/cisco/apic/mechanism_apic.py

```diff
@@ -312,11 +312,39 @@ def bind_port(self, context):
                       vnic_type)
             return
 
-        # Attempt to bind ports for DVS agents for nova-compute daemons
-        # first. This allows having network agents (dhcp, metadata)
-        # that typically run on a network node using an OpFlex agent to
-        # co-exist with nova-compute daemons for ESX, which host DVS agents.
         if context.current['device_owner'].startswith('compute:'):
+            # enforce the secure port binding rules if possible
+            if context.current['device_id']:
+                vm = nova_client.NovaClient().get_server(
```

Comment:
Please avoid calling into Nova for cases when there are no restrictions (the default case)